routes/login.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+var nconf = require('nconf');
+var express = require('express');
+var bodyParser = require('body-parser')
+var router = express.Router();
+
+// create application/json parser
+var jsonParser = bodyParser.json()
+// create application/x-www-form-urlencoded parser
+var urlencodedParser = bodyParser.urlencoded({ extended: false })
+
+var redirectBack = function(req, res) {
+    var redirectAfterLogin = req.session.redirectAfterLogin ? req.session.redirectAfterLogin : '/';
+    delete req.session.redirectAfterLogin;
+    res.redirect(redirectAfterLogin);
+};
+
+router.get('/', function(req, res) {
+    res.status(200);
+    res.render('login', {
+        username: req.session.username
+    });
+});
+
+router.post('/', urlencodedParser, function(req, res) {
+    if (!req.body) return res.sendStatus(400)
```

Comment:
Always use `{}` and put `if` body on a new line. Makes it easier to read and debug (adding breakpoints).

Also, this line and some others in the file are missing semicolons.